tests/kafkatest/tests/core/log_dir_failure_test.py

```diff
@@ -0,0 +1,179 @@
+                                      [config_property.LOG_ROLL_TIME_MS, "3000"]
+                                  ])
+
+        self.producer_throughput = 1000
+        self.num_producers = 1
+        self.num_consumers = 1
+
+    def setUp(self):
+        self.zk.start()
+
+    def min_cluster_size(self):
+        """Override this since we're adding services outside of the constructor"""
+        return super(LogDirFailureTest, self).min_cluster_size() + self.num_producers * 2 + self.num_consumers * 2
+
+    @cluster(num_nodes=9)
+    @matrix(bounce_broker=[False, True], broker_type=["leader", "follower"], security_protocol=["PLAINTEXT"])
+    def test_replication_with_disk_failure(self, bounce_broker, security_protocol, broker_type):
+        """Replication tests.
+        These tests verify that replication provides simple durability guarantees by checking that data acked by
+        brokers is still available for consumption in the face of various failure scenarios.
+
+        Setup: 1 zk, 3 kafka nodes, 1 topic with partitions=3, replication-factor=3, and min.insync.replicas=2
+
+            - Produce messages in the background
+            - Consume messages in the background
+            - Drive broker failures (shutdown, or bounce repeatedly with kill -15 or kill -9)
+            - When done driving failures, stop producing, and finish consuming
+            - Validate that every acked message was consumed
+        """
+
+        self.kafka.security_protocol = security_protocol
+        self.kafka.interbroker_security_protocol = security_protocol
+        self.kafka.start()
+
+        try:
+            # Initialize producer/consumer for topic1
+            self.producer = VerifiableProducer(self.test_context, self.num_producers, self.kafka, self.topic1,
+                                               throughput=self.producer_throughput)
+            self.consumer = ConsoleConsumer(self.test_context, self.num_consumers, self.kafka, self.topic1, group_id="test-consumer-group-1",
+                                            new_consumer=False, consumer_timeout_ms=60000, message_validator=is_int)
+            self.start_producer_and_consumer()
+
+            # Get a replica of the partition of topic1 and make its first log directory offline by changing the log dir's permission.
+            # We assume that partition of topic1 is created in the first log directory of respective brokers.
+            broker_node = select_node(self, broker_type, self.topic1)
+            broker_idx = self.kafka.idx(broker_node)
+            assert broker_idx in self.kafka.isr_idx_list(self.topic1), \
+                   "Leader node %d should be in isr set %s" % (broker_idx, str(self.kafka.isr_idx_list(self.topic1)))
+
+            self.logger.debug("Making log dir %s inaccessible" % (KafkaService.DATA_LOG_DIR_1))
+            cmd = "sudo chmod a-rw %s -R" % (KafkaService.DATA_LOG_DIR_1)
```

Comment:
Yes.. I think it is needed. Below is the error if this command is executed without sudo.

```
Traceback (most recent call last):
  File "/export/apps/python/2.7.13/lib/python2.7/site-packages/ducktape/tests/runner_client.py", line 123, in run
    data = self.run_test()
  File "/export/apps/python/2.7.13/lib/python2.7/site-packages/ducktape/tests/runner_client.py", line 176, in run_test
    return self.test_context.function(self.test)
  File "/export/apps/python/2.7.13/lib/python2.7/site-packages/ducktape/mark/_mark.py", line 321, in wrapper
    return functools.partial(f, *args, **kwargs)(*w_args, **w_kwargs)
  File "/home/dolin/research/kafka/tests/kafkatest/tests/core/log_dir_failure_test.py", line 133, in test_replication_with_disk_failure
    broker_node.account.ssh(cmd, allow_fail=False)
  File "/export/apps/python/2.7.13/lib/python2.7/site-packages/ducktape/cluster/remoteaccount.py", line 253, in ssh
    raise RemoteCommandError(self, cmd, exit_status, stderr.read())
RemoteCommandError: vagrant@worker9: Command 'chmod a-rw /mnt/kafka-data-logs-1 -R' returned non-zero exit status 1. Remote error message: chmod: cannot read directory ‘/mnt/kafka-data-logs-1’: Permission denied
```

I also tried to execute the command with and without sudo by logging into the vagrant node. Here is what I found:

```
vagrant@worker5:~$ ls -ld /mnt/kafka-data-logs-1/
d--x--x--x 3 vagrant vagrant 4096 Jul 21 03:19 /mnt/kafka-data-logs-1/
vagrant@worker5:~$ whoami 
vagrant
vagrant@worker5:~$ chmod a-rw /mnt/kafka-data-logs-1/ -R 
chmod: cannot read directory ‘/mnt/kafka-data-logs-1/’: Permission denied
vagrant@worker5:~$ sudo chmod a-rw /mnt/kafka-data-logs-1/ -R 
vagrant@worker5:~$ 
```